// neto-api/dist/modules/category.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
const update_1 = require("./methods/update");
class AddCategory extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddCategory", schema: "Category" }));
    }
}
exports.AddCategory = AddCategory;
class GetCategory extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetCategory" }));
    }
}
exports.GetCategory = GetCategory;
class UpdateCategory extends update_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "UpdateCategory", schema: "Category" }));
    }
}
exports.UpdateCategory = UpdateCategory;
exports.methods = {
    add: (data) => {
        return new AddCategory(data);
    },
    get: (filter) => {
        return new GetCategory(filter);
    },
    update: (data) => {
        return new UpdateCategory(data);
    }
};
