// neto-api/dist/modules/content.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
const update_1 = require("./methods/update");
class AddContent extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddContent", schema: "Content" }));
    }
}
exports.AddContent = AddContent;
class GetContent extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetContent" }));
    }
}
exports.GetContent = GetContent;
class UpdateContent extends update_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "UpdateContent", schema: "Content" }));
    }
}
exports.UpdateContent = UpdateContent;
exports.methods = {
    add: (data) => {
        return new AddContent(data);
    },
    get: (filter) => {
        return new GetContent(filter);
    },
    update: (data) => {
        return new UpdateContent(data);
    }
};
