// neto-api/dist/modules/customer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const shared_1 = require("../shared");
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
const update_1 = require("./methods/update");
class AddCustomer extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddCustomer", schema: "Customer" }));
    }
}
exports.AddCustomer = AddCustomer;
class GetCustomer extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetCustomer" }));
    }
}
exports.GetCustomer = GetCustomer;
class UpdateCustomer extends update_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "UpdateCustomer", schema: "Customer" }));
    }
}
exports.UpdateCustomer = UpdateCustomer;
class AddCustomerLog {
    constructor(data) {
        this.addModule = new add_1.default(data);
        return this;
    }
    addLog(data) {
        this.addModule.add(data);
        return this;
    }
    exec({ debug = false } = {}) {
        let body = {
            CustomerLogs: {
                CustomerLog: this.addModule.data
            }
        };
        let req = { action: "AddCustomerLog", body: body };
        if (debug)
            return req;
        return shared_1.postApi(req);
    }
}
exports.AddCustomerLog = AddCustomerLog;
class UpdateCustomerLog {
    constructor(data) {
        this.updateModule = new update_1.default(data);
        return this;
    }
    updateLog(data) {
        this.updateModule.update(data);
        return this;
    }
    exec({ debug = false } = {}) {
        let body = {
            CustomerLogs: {
                CustomerLog: this.updateModule.data
            }
        };
        let req = { action: "UpdateCustomerLog", body: body };
        if (debug)
            return req;
        return shared_1.postApi(req);
    }
}
exports.UpdateCustomerLog = UpdateCustomerLog;
exports.methods = {
    add: (data) => {
        return new AddCustomer(data);
    },
    get: (filter) => {
        return new GetCustomer(filter);
    },
    update: (data) => {
        return new UpdateCustomer(data);
    },
    addLog: (data) => {
        return new AddCustomerLog(data);
    },
    updateLog: (data) => {
        return new UpdateCustomerLog(data);
    }
};
