// neto-api/dist/modules/item.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
const update_1 = require("./methods/update");
class AddItem extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddItem", schema: "Item" }));
    }
}
exports.AddItem = AddItem;
class GetItem extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetItem" }));
    }
}
exports.GetItem = GetItem;
class UpdateItem extends update_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "UpdateItem", schema: "Item" }));
    }
}
exports.UpdateItem = UpdateItem;
exports.methods = {
    add: (data) => {
        return new AddItem(data);
    },
    get: (filter) => {
        return new GetItem(filter);
    },
    update: (data) => {
        return new UpdateItem(data);
    }
};
