// neto-api/dist/modules/order.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
const update_1 = require("./methods/update");
class AddOrder extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddOrder", schema: "Order" }));
    }
}
exports.AddOrder = AddOrder;
class GetOrder extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetOrder" }));
    }
}
exports.GetOrder = GetOrder;
class UpdateOrder extends update_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "UpdateOrder", schema: "Order" }));
    }
}
exports.UpdateOrder = UpdateOrder;
exports.methods = {
    add: (data) => {
        return new AddOrder(data);
    },
    get: (filter) => {
        return new GetOrder(filter);
    },
    update: (data) => {
        return new UpdateOrder(data);
    }
};
