// neto-api/dist/modules/payment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const shared_1 = require("../shared");
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
class AddPayment extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddPayment", schema: "Payment" }));
    }
}
exports.AddPayment = AddPayment;
class GetPayment extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetPayment" }));
    }
}
exports.GetPayment = GetPayment;
class GetPaymentMethods {
    constructor() {
        return this;
    }
    exec({ debug = false } = {}) {
        let req = { action: "GetPaymentMethods", body: {} };
        if (debug)
            return req;
        return shared_1.postApi(req);
    }
}
exports.GetPaymentMethods = GetPaymentMethods;
exports.methods = {
    getMethods: () => {
        return new GetPaymentMethods();
    },
    get: (filter) => {
        return new GetPayment(filter);
    },
    add: (data) => {
        return new AddPayment(data);
    }
};
