// neto-api/dist/modules/supplier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
const update_1 = require("./methods/update");
class AddSupplier extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddSupplier", schema: "Supplier" }));
    }
}
exports.AddSupplier = AddSupplier;
class GetSupplier extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetSupplier" }));
    }
}
exports.GetSupplier = GetSupplier;
class UpdateSupplier extends update_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "UpdateSupplier", schema: "Supplier" }));
    }
}
exports.UpdateSupplier = UpdateSupplier;
exports.methods = {
    add: (data) => {
        return new AddSupplier(data);
    },
    get: (filter) => {
        return new GetSupplier(filter);
    },
    update: (data) => {
        return new UpdateSupplier(data);
    }
};
