// neto-api/dist/modules/voucher.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
const update_1 = require("./methods/update");
class AddVoucher extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddVoucher", schema: "Voucher" }));
    }
}
exports.AddVoucher = AddVoucher;
class GetVoucher extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetVoucher" }));
    }
}
exports.GetVoucher = GetVoucher;
class UpdateVoucher extends update_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "UpdateVoucher", schema: "Voucher" }));
    }
}
exports.UpdateVoucher = UpdateVoucher;
exports.methods = {
    add: (data) => {
        return new AddVoucher(data);
    },
    get: (filter) => {
        return new GetVoucher(filter);
    },
    update: (data) => {
        return new UpdateVoucher(data);
    }
};
