// neto-api/dist/modules/warehouse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const add_1 = require("./methods/add");
const get_1 = require("./methods/get");
const update_1 = require("./methods/update");
class AddWarehouse extends add_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "AddWarehouse", schema: "Warehouse" }));
    }
}
exports.AddWarehouse = AddWarehouse;
class GetWarehouse extends get_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "GetWarehouse" }));
    }
}
exports.GetWarehouse = GetWarehouse;
class UpdateWarehouse extends update_1.default {
    exec(params = {}) {
        return super.exec(Object.assign(Object.assign({}, params), { action: "UpdateWarehouse", schema: "Warehouse" }));
    }
}
exports.UpdateWarehouse = UpdateWarehouse;
exports.methods = {
    add: (data) => {
        return new AddWarehouse(data);
    },
    get: (filter) => {
        return new GetWarehouse(filter);
    },
    update: (data) => {
        return new UpdateWarehouse(data);
    }
};
