// neto-api/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const cartModule = require("./modules/cart");
const categoryModule = require("./modules/category");
const contentModule = require("./modules/content");
const currencyModule = require("./modules/currency");
const customerModule = require("./modules/customer");
const itemModule = require("./modules/item");
const orderModule = require("./modules/order");
const paymentModule = require("./modules/payment");
const rmaModule = require("./modules/rma");
const shippingModule = require("./modules/shipping");
const supplierModule = require("./modules/supplier");
const voucherModule = require("./modules/voucher");
const warehouseModule = require("./modules/warehouse");
const shared_1 = require("./shared");
class NetoAPI {
    constructor(options) {
        this.item = itemModule.methods;
        this.order = orderModule.methods;
        this.customer = customerModule.methods;
        this.category = categoryModule.methods;
        this.content = contentModule.methods;
        this.cart = cartModule.methods;
        this.supplier = supplierModule.methods;
        this.voucher = voucherModule.methods;
        this.warehouse = warehouseModule.methods;
        this.rma = rmaModule.methods;
        this.shipping = shippingModule.methods;
        this.currency = currencyModule.methods;
        this.payment = paymentModule.methods;
        shared_1.init(options);
    }
}
exports.NetoAPI = NetoAPI;
